### ISSUE TITLE
error to send bulk of actions - bulk request timed out

### ISSUE THREAD

**Issue by rolandah** [logstash-plugins/logstash-output-elasticsearch#103]
I am facing weird issues with my ELK lately:
- logstash crashes for no reason; java keeps running at relevant port but it does not forward logs to output/elasticsearch
- logstash-forwarder gives i/o errors which is expected given the above error on logstash service..
- if i use /etc/init.d/logstash restart which is a script provided by logstash deb package itself ; it fails to stop the previous running pid and results to this error
The error reported is: 
  Address already in use - bind - Address already in use

at the same time ;elasticsearch is running behind nginx on a different ec2 instance without any problem... 

the below is the most recent error; 

{:timestamp=>"2015-04-11T10:19:44.509000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at testing.intranet : Read timed out", :level=>:error}
{:timestamp=>"2015-04-11T10:22:51.059000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at testing.intranet : Read timed out", :level=>:error}
{:timestamp=>"2015-04-11T10:22:51.060000+0000", :message=>"Failed to flush outgoing items", :outgoing_count=>2475, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:61:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:224:in `call_once'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:127:in `code'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch/protocol.rb:82:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:413:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:438:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:402:in `receive'", "/opt/logstash/lib/logstash/outputs/base.rb:88:in `handle'", "(eval):1906:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/lib/logstash/pipeline.rb:279:in `output'", "/opt/logstash/lib/logstash/pipeline.rb:235:in `outputworker'", "/opt/logstash/lib/logstash/pipeline.rb:163:in `start_outputs'"], :level=>:warn}
{:timestamp=>"2015-04-11T10:23:26.290000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at testing.intranet : Read timed out", :level=>:error}
{:timestamp=>"2015-04-11T10:23:26.291000+0000", :message=>"Failed to flush outgoing items", :outgoing_count=>2599, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:61:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:224:in `call_once'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:127:in `code'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch/protocol.rb:82:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:413:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:438:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:402:in `receive'", "/opt/logstash/lib/logstash/outputs/base.rb:88:in `handle'", "(eval):1906:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/lib/logstash/pipeline.rb:279:in `output'", "/opt/logstash/lib/logstash/pipeline.rb:235:in `outputworker'", "/opt/logstash/lib/logstash/pipeline.rb:163:in `start_outputs'"], :level=>:warn}
{:timestamp=>"2015-04-11T10:24:05.002000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at testing.intranet : Read timed out", :level=>:error}
{:timestamp=>"2015-04-11T10:37:14.720000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at testing.intranet : Read timed out", :level=>:error}
{:timestamp=>"2015-04-11T10:37:14.721000+0000", :message=>"Failed to flush outgoing items", :outgoing_count=>2356, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:61:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:224:in `call_once'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.3.5-java/lib/manticore/response.rb:127:in `code'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch/protocol.rb:82:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:413:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:438:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash/outputs/elasticsearch.rb:402:in `receive'", "/opt/logstash/lib/logstash/outputs/base.rb:88:in `handle'", "(eval):1906:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/lib/logstash/pipeline.rb:279:in `output'", "/opt/logstash/lib/logstash/pipeline.rb:235:in `outputworker'", "/opt/logstash/lib/logstash/pipeline.rb:163:in `start_outputs'"], :level=>:warn}
{:timestamp=>"2015-04-11T10:37:54.493000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at testing.intranet : Read timed out", :level=>:error}

**Comment by suyograo** [comment]
Related issue: https://github.com/elastic/logstash/issues/3049

**Comment by elisiano** [comment]
I have a problem similar to this. My setup is only for testing at the current time, hence running on my laptop. 
In my case I'm running a local couchdb and a local elasticsearch instance.
Using couchdb_changes as input (new in logstash 1.5.X) and elasticsearch as output (with some filters to tag documents).

The behavior is similar except logstash does not crash (although once in this state it does not update anymore).
```
Got error to send bulk of actions to elasticsearch server at 127.0.0.1 : Read timed out {:level=>:error}
Failed to flush outgoing items {:outgoing_count=>1278, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61:in `call'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225:in `call_once'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128:in `code'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch/protocol.rb:100:in `bulk'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:437:in `submit'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:462:in `flush'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:426:in `receive'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0.rc3-java/lib/logstash/outputs/base.rb:88:in `handle'", "(eval):221:in `output_func'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0.rc3-java/lib/logstash/pipeline.rb:244:in `outputworker'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0.rc3-java/lib/logstash/pipeline.rb:166:in `start_outputs'"], :level=>:warn}
```

Here's the relevant logstash config bits:
```
input {
        couchdb_changes {
                db => "<MY_DB>"
        }
}
filter {
...some manipulation...
}
output {
[...some conditional to output the to different elasticsearch outputs with different 'document_type'...]
        elasticsearch {
                action => "%{[@metadata][action]}"
                        document_id => "%{[@metadata][_id]}"
                        index => '<CONDITIONAL_INDEX>'
                        document_type => '<CONDITIONAL_TYPE>'
                        host => "127.0.0.1"
                        protocol => "http"
        }
}
```

In the elasticsearch logs I start seeing this after logstash starts failing:
```
[2015-05-03 11:20:30,186][DEBUG][action.bulk              ] [L-99] observer: timeout notification from cluster service. timeout setting [1m], time since start [1m]
[2015-05-03 11:20:30,186][DEBUG][action.bulk              ] [L-99] observer: timeout notification from cluster service. timeout setting [1m], time since start [1m]
[2015-05-03 11:20:30,186][DEBUG][action.bulk              ] [L-99] observer: timeout notification from cluster service. timeout setting [1m], time since start [1m]
```

**Comment by elisiano** [comment]
Actually, if I let it run long enough it eventually crashes. I'm using logstash-1.5.0-rc3

```
[...a log of these error message cut...]
too many attempts at sending event. dropping: 2012-09-29T07:58:14.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-09-29T18:37:49.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-05-19T13:21:22.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-05-19T14:21:34.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-05-19T14:41:54.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-05-30T07:26:21.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-05-30T07:26:59.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
too many attempts at sending event. dropping: 2012-05-30T07:27:48.000Z %{host} %{message} {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"592", :method=>"retry_flush"}
Got error to send bulk of actions to elasticsearch server at 127.0.0.1 : Read timed out {:level=>:error, :file=>"logstash/outputs/elasticsearch.rb", :line=>"464", :method=>"flush"}
Shifting current elasticsearch client {:level=>:debug, :file=>"logstash/outputs/elasticsearch.rb", :line=>"468", :method=>"flush"}
Switched current elasticsearch client to #0 at 127.0.0.1 {:level=>:debug, :file=>"logstash/outputs/elasticsearch.rb", :line=>"518", :method=>"shift_client"}
Failed to flush outgoing items {:outgoing_count=>2257, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61:in `call'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225:in `call_once'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128:in `code'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch/protocol.rb:100:in `bulk'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:437:in `submit'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:462:in `flush'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:112:in `buffer_initialize'", "org/jruby/RubyKernel.java:1507:in `loop'", "/Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:110:in `buffer_initialize'"], :level=>:warn, :file=>"stud/buffer.rb", :line=>"231", :method=>"buffer_flush"}
Sending bulk of actions to client[0]: 127.0.0.1 {:level=>:debug, :file=>"logstash/outputs/elasticsearch.rb", :line=>"461", :method=>"flush"}
Manticore::Timeout: Read timed out
       initialize at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35
             call at org/jruby/RubyProc.java:271
             call at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61
        call_once at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225
             code at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128
  perform_request at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50
             call at org/jruby/RubyProc.java:271
  perform_request at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187
  perform_request at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33
  perform_request at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115
             bulk at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80
             bulk at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch/protocol.rb:100
           submit at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:437
      retry_flush at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:599
         register at /Users/epetrini/Projects/fabriccare/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:380
```

**Comment by horkhe** [comment]
Experiencing the same error. Logstash stops pushing messages to elasticsearch with:

```
{:timestamp => "2015-05-20T04:34:15.185000+0000", 
 :message => "Failed to flush outgoing items", 
 :outgoing_count => 100, 
 :exception => #<Manticore::Timeout: Read timed out>,
 :backtrace => [
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35:in `initialize'",
      "org/jruby/RubyProc.java:271:in `call'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61:in `call'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225:in `call_once'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128:in `code'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'",
      "org/jruby/RubyProc.java:271:in `call'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch/protocol.rb:100:in `bulk'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:437:in `submit'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:436:in `submit'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:462:in `flush'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:460:in `flush'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'",
      "org/jruby/RubyHash.java:1341:in `each'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:426:in `receive'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0.rc3-java/lib/logstash/outputs/base.rb:88:in `handle'",
      "(eval):247:in `output_func'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0.rc3-java/lib/logstash/pipeline.rb:244:in `outputworker'",
      "/opt/logstash/server/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0.rc3-java/lib/logstash/pipeline.rb:166:in `start_outputs'"],
 :level => :warn}
{:timestamp => "2015-05-20T04:34:26.213000+0000",
 :message => "Got error to send bulk of actions to elasticsearch server at 127.0.0.1 : Read timed out",
 :level => :error}
```

But we are screwed for that happens in production. So far we just restart logstash instances to make it resume.

As you can see we are using 1.5.0.rc3. Will try updating to 1.5.0 release and see if it helps.

**Comment by colinsanders** [comment]
FYI I'm experiencing the same problem with the 1.5.0 release.

**Comment by horkhe** [comment]
@colinsanders thanks for information. Then I guess there is nothing to do but rollback to v1.4.x

**Comment by socifi** [comment]
We have the same problem with 1.5.0...

**Comment by talevy** [comment]
@elisiano @horkhe @colinsanders @socifi 

would you all mind sharing the relevant parts of your logstash configuration, how you have 
logstash and elasticsearch running (e.g. how many nodes, protocol), and which version of elasticsearch you are running?

I am having trouble recreating this issue

**Comment by elisiano** [comment]
@talevy: in my case it was only a test setup, 1 local node on my laptop.
The relevant logstash config is in one of my previous comments (I tried some variations, even passing through a redis server, but nothing changed, the bottleneck seems the elasticsearch output).

I tested it with logstash 1.5 rc3 and rc4 (same results).
The elasticsearch version tested was 1.5.1.

**Comment by talevy** [comment]
@elisiano does this start happening after a long time? or very soon after you initiate the pipeline? How is your system resource usage at this time? are you indexing small or large documents?

**Comment by elisiano** [comment]
No, after few minutes. The exact amount of time varies every time (and also depends on how the elasticsearch instance is sized - i.e.: ES_HEAP_SIZE).
I'm indexing small documents (few kilobytes).
I also tried with the final 1.5.0.

The behavior is not every time the same, sometimes logstash shutdowns, and sometimes it hangs there but I consistently get the usual `Got error to send bulk of actions to elasticsearch server at 127.0.0.1 : Read timed out {:level=>:error}`, it keeps going for a while (I'm using the stdout dots codec so I see records being processed) and then after a short while I get this:
```
Exception in thread "<couchdb_changes" java.lang.UnsupportedOperationException
	at java.lang.Thread.stop(Thread.java:869)
	at org.jruby.RubyThread.exceptionRaised(RubyThread.java:1221)
	at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:112)
	at java.lang.Thread.run(Thread.java:745)
Exception in thread "Ruby-0-Thread-13: /Users/epetrini/Projects/fabriccare/vendor/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:92" java.lang.UnsupportedOperationException
	at java.lang.Thread.stop(Thread.java:869)
	at org.jruby.RubyThread.exceptionRaised(RubyThread.java:1221)
	at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:112)
	at java.lang.Thread.run(Thread.java:745)
```

The resources of the system are never above 75% (on the CPU side, the RAM depends how much I allocate for ES, but it never maxes out).

**Comment by talevy** [comment]
@elisiano that looks like a couchdb exception. are you experiencing this ES timeout when not using couchdb as well?

**Comment by spolk01** [comment]
same here

"Got error to send bulk of actions to elasticsearch server at 172.19.2.10 : Read timed out", :level=>:error}

I am on logstash-1.5.0-1 from rpm.

I can't switch back to 1.4.2 as I had another out-of-memory problem that 1.5 supposed to fix so neither one works for me....

**Comment by colinsanders** [comment]
@talevy: apologies for my late reply. 
1 local node on my laptop (Mac)
Using ES 1.5.2 and LS1.5.0

I have about 12450 json files to index. 
I got the error message almost immediately. 
The number of files sent to ES was varying every time.

When I changed the maxfiles and maxfilesperproc (way over the number of files) on my Mac, I did no longer get the error message. But only 10191 files are index. At every try.
However I do get  a warning: "failed to open /Users/xyz/abcd7.json: Permission denied"

Since all my input file end with a number eg, abcd1.json, abcd2.json etc. I copied the config file into 11 files.  e.g.     path => "/Users/xyz/*2.json"
When I do this, I can import all 12450 files into ES flawless. This worked  already before I changed the maxfiles.

I hope this helps.

Regards,

Colin

**Comment by wa15h** [comment]
Same issue here....

Manticore::Timeout: Read timed out
       initialize at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35
             call at org/jruby/RubyProc.java:271
             call at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61
        call_once at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225
             code at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128
  perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/transport/http/manticore.rb:71
             call at org/jruby/RubyProc.java:271
  perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/transport/base.rb:187
  perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/transport/http/manticore.rb:54
  perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/client.rb:115
             bulk at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.9/lib/elasticsearch/api/actions/bulk.rb:80
             bulk at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch/protocol.rb:100
           submit at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:437
           submit at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:436
      retry_flush at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:598
         register at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:380

{:timestamp=>"2015-05-22T14:24:17.593000-0500", :message=>"Failed to flush outgoing items", :outgoing_count=>11957, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225:in `call_once'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128:in `code'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/transport/http/manticore.rb:71:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/transport/http/manticore.rb:54:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.9/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.9/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch/protocol.rb:100:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:437:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:436:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:461:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.5-java/lib/logstash/outputs/elasticsearch.rb:460:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:112:in `buffer_initialize'", "org/jruby/RubyKernel.java:1507:in `loop'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:110:in `buffer_initialize'"], :level=>:warn}

logstash-1.5.0-1.noarch
elasticsearch-1.5.2-1.noarch

**Comment by rholloway** [comment]
+1 same issue. Running several on EC2 instances in AWS (m3.medium)  behind ELB

Logstash-1.5.0-1
Elasticsearch 1.4.5

I did not encounter this issue until upgrading to logstash 1.5.0 last week, although I upgraded because the prior version of logstash I was on was not playing well with logstash-forwarder for some other reason...

**Comment by suyograo** [comment]
Update: @talevy was able to reproduce this behavior using a network simulator and increasing his network latency. This maybe related to timeout settings in manticore.

**Comment by bojbo** [comment]
ES 1.5.2
LS 1.5.1

When i install elasticsearch's plugin 'watcher', i got these errors in logstash.log


{:timestamp=>"2015-05-28T11:31:57.064000+0800", :message=>"Failed to install template: waited for [30s]", :level=>:error}
{:timestamp=>"2015-05-28T11:32:58.282000+0800", :message=>"Got error to send bulk of actions to elasticsearch server at 115.182.69.200 : blocked by: [SERVICE_UNAVAILABLE/1/state not recovered / initialized];[SERVICE_UNAVAILABLE/2/no master];", :level=>:error}
{:timestamp=>"2015-05-28T11:32:58.287000+0800", :message=>"Failed to flush outgoing items", :outgoing_count=>49, :exception=>org.elasticsearch.cluster.block.ClusterBlockException: blocked by: [SERVICE_UNAVAILABLE/1/state not recovered / initialized];[SERVICE_UNAVAILABLE/2/no master];, :backtrace=>["org.elasticsearch.cluster.block.ClusterBlocks.globalBlockedException(org/elasticsearch/cluster/block/ClusterBlocks.java:151)", "org.elasticsearch.cluster.block.ClusterBlocks.globalBlockedRaiseException(org/elasticsearch/cluster/block/ClusterBlocks.java:141)", "org.elasticsearch.action.bulk.TransportBulkAction.executeBulk(org/elasticsearch/action/bulk/TransportBulkAction.java:210)", "org.elasticsearch.action.bulk.TransportBulkAction.access$000(org/elasticsearch/action/bulk/TransportBulkAction.java:73)", "org.elasticsearch.action.bulk.TransportBulkAction$1.onFailure(org/elasticsearch/action/bulk/TransportBulkAction.java:148)", "org.elasticsearch.action.support.TransportAction$ThreadedActionListener$2.run(org/elasticsearch/action/support/TransportAction.java:137)", "java.util.concurrent.ThreadPoolExecutor.runWorker(java/util/concurrent/ThreadPoolExecutor.java:1145)", "java.util.concurrent.ThreadPoolExecutor$Worker.run(java/util/concurrent/ThreadPoolExecutor.java:615)", "java.lang.Thread.run(java/lang/Thread.java:745)"], :level=>:warn}

**Comment by ralphm** [comment]
@jsvd thanks for that change, I am going to test this out as soon as possible. However, why does the case with the timeout make logstash stall? It seems that bulk requests that fail in this way are not retried as with other failures and silently dropped, but also cause logstash to a grinding halt. Ideas on that?

@bojbo: I'm not sure if that log trace has anything to do with this particular issue.

**Comment by jsvd** [comment]
@ralphm I'm unsure if the timeout issue is connected with logstash stalling, actually. #162 will revert the timeout behaviour to 1.4.2 state, and from there we can carry on analyzing the hanging scenarios..

**Comment by landonix** [comment]
I made the suggested changes to disable timeouts, but the issue persists after a full LS restart. ES 1.5.2, LS 1.5.0-1, LSF 0.40

{:timestamp=>"2015-05-28T19:50:49.561000+0000", :message=>"Got error to send bulk of actions to elasticsearch server at localhost : Read timed out", :level=>:error}
{:timestamp=>"2015-05-28T19:50:49.562000+0000", :message=>"Failed to flush outgoing items", :outgoing_count=>4240, :exception=>#<Manticore::Timeout: Read timed out>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:35:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:61:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:225:in `call_once'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.1-java/lib/manticore/response.rb:128:in `code'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:50:in `perform_request'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/transport/http/manticore.rb:33:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.7/lib/elasticsearch/transport/client.rb:115:in `perform_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.7/lib/elasticsearch/api/actions/bulk.rb:80:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch/protocol.rb:102:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:437:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:436:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:462:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:460:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.4-java/lib/logstash/outputs/elasticsearch.rb:426:in `receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/outputs/base.rb:88:in `handle'", "(eval):274:in `output_func'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/pipeline.rb:244:in `outputworker'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/pipeline.rb:166:in `start_outputs'"], :level=>:warn}

**Comment by landonix** [comment]
I've made these changes and will continue to monitor on a very active system.

**Comment by ralphm** [comment]
@jsvd: right. I am trying to dig through some code to find possible causes for Logstash getting stuck on this. As far as I can tell, it is `Stud::Buffer` that is logging the exception raised in `LogStash::Outputs::ElasticSearch::flush`. `Stud::Buffer` then sleeps for 1 second and issues a retry. I assume that means that the output plugin is then blocked for that one second, thereby blocking the entire pipeline, but I haven't yet determined if that retry actually succeeds or that this is dropped on the floor.

The problem is hard to debug for me because it can take hours for this to occur in my setup.

**Comment by ralphm** [comment]
@jsvd: I collected a stack trace for a hung process: https://gist.github.com/ralphm/e14a18808a622b565744.

**Comment by hrchu** [comment]
I'm experiencing the same problem with the 1.5.0-1 release. Does anyone know that the message "Got error to send bulk of actions to elasticsearch server  : Read timed out" means that the log is loss, or logstash will retry to send?

**Comment by jsvd** [comment]
Timeouts have been disabled in the manticore library the elasticsearch output uses to speak http in https://github.com/logstash-plugins/logstash-output-elasticsearch/commit/5c0a328e002c7ff724f835e9de15fb350c597535
This reverts logstash elasticsearch output on http behaviour to the same as 1.4.2

**Comment by innovia** [comment]
im having the same issue, running ElasticSearch 1.6.0 and logstash 1.5.0

getting stuck at:
Sending bulk of actions to client[0]: 10.10.10.39:9300 {:level=>:debug, :file=>"logstash/outputs/elasticsearch.rb", :line=>"461", :method=>"flush"}

strace does not really help so much, i see egain errors, and something about connection refuse

im running my ES cluster on private subnet behind AWS ELB

can't find a fix

**Comment by socifi** [comment]
@innovia I had same error with all releases (1.4-1.5.1) if *http* protocol was set...

It might help: change logstash <-> es transport protocol from *http* to *node*.

For example:
```
elasticsearch {
    cluster => "my-cluster"
    host => ["10.0.0.X","10.0.0.Y"]
    protocol => "node"
    port => 9300
    index => "my-index-%{+YYYY.MM.dd}"
}
```
The port 9300 must be available on each side.. The logstash instance will be displayed within ES cluster nodes... (CPU on logstash goes down from 40-50% to <5% and we have no more missing or duplicated logs)
more info: https://www.elastic.co/guide/en/elasticsearch/guide/current/_talking_to_elasticsearch.html

**Comment by innovia** [comment]
this is exactly my config, and i do see logs getting indexes for a while until it stops 

i can see the instance of logstash on es cluster

when i restart the process of logstash its indexing again, but after a while its stops

**Comment by socifi** [comment]
How many logs per sec? And instance type? Ive used r3.large for ES via AWS LB and m3.medium for LS with 10-100l/s - no luck via http. After few minutes it went down. (CPU/heap was ok, but exception occurred)
Now Im on half-sized instances in Azure and everything works lawlessly with "node" protocol. (again no luck with http). 
If you already have node protocol set, im sorry but can't help you more..

logstash: 1.5.1-1
java: Java(TM) SE Runtime Environment (build 1.8.0_45-b14)

**Comment by innovia** [comment]
2 instances of ES (c3.xlarge) behind elb on private subnet (VPC) no access to internet
1 instance of logstash m3.large on public subnet (internet facing)


logstash 1.5.1
es 1.6.0

**Comment by drocsid** [comment]
{:timestamp=>"2015-07-10T13:26:16.825000-0600", :message=>"Failed to flush outgoing items", :outgoing_count=>1, :exception=>#<NameError: no method 'type' for arguments (org.jruby.RubyArray) on Java::OrgElasticsearchActionIndex::IndexRequest>, :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch/protocol.rb:261:in `build_request'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch/protocol.rb:222:in `bulk'", "org/jruby/RubyArray.java:1613:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch/protocol.rb:221:in `bulk'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch.rb:466:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch.rb:465:in `submit'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch.rb:490:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch.rb:489:in `flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:219:in `buffer_flush'", "org/jruby/RubyHash.java:1341:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:216:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:193:in `buffer_flush'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/buffer.rb:159:in `buffer_receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.2.8-java/lib/logstash/outputs/elasticsearch.rb:455:in `receive'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.1-java/lib/logstash/outputs/base.rb:88:in `handle'", "(eval):520:in `output_func'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.1-java/lib/logstash/pipeline.rb:243:in `outputworker'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.1-java/lib/logstash/pipeline.rb:165:in `start_outputs'"], :level=>:warn}

Feels like there is no QA for these logstash / elasticsearch releases. If you search bulk on github, you will see this issue all over the place.

**Comment by talevy** [comment]
@drocsid 

For these cases, there are numerous different scenarios that one would lead to  unsuccessfully indexing into Elasticsearch. Network delays, messages with incorrect mappings, etc. Your error suggests that you are using the transport/node protocol and running into a problem when setting the `type`.

would you mind sharing your elasticsearch plugin configuration? (no need to share the hosts or any other sensitive information)

**Comment by drocsid** [comment]
elasticsearch {
      host => '127.0.0.1'
      cluster => logstash
      bind_port => 9302
    }

**Comment by talevy** [comment]
@drocsid hmm. are you receiving this error every time? what is the nature of your input?

**Comment by drocsid** [comment]
I see my two logstash servers deliver these messages after running for a few days, and no more content delivered.

**Comment by talevy** [comment]
@drocsid I see. would you mind filing a new issue with this? It is hard to say that this is a timeout issue because I feel some context is missing from the stacktrace you provided.

It would be much more verbose, but running logstash with `--debug` flag set may yield more information when the error occurs.

**Comment by drocsid** [comment]
Hi @talevy , I previously had it running with the --debug flag, but thought the issue went away. I'm going to try a few things myself, and will file an issue soon if I can't resolve it with my testing. Still I feel like QA.

**Comment by drocsid** [comment]
@talevy FWIW: here is the configuration, with some of the type and fields modified:

input {

  lumberjack {
    port => 5544
    type => "lumberjack"
    ssl_certificate => "/etc/logstash/lumberjack.crt"
    ssl_key => "/etc/logstash/lumberjack.key"
  }
}

filter {
  if [type]==z-prod" {
    grok {
    match => [ "message", "%{TIMESTAMP_ISO8601:timestamp}%{SPACE}(?<log_level>WARN|ERROR|FATAL)%{SPACE}%{JAVACLASS:java_class_path}" ]
      add_field => {
      "service" => "z"
      "cluster" => "z"
      }
    }
    date {
      match => [ "timestamp" , "YYYY-MM-dd HH:mm:ss,SSS" ]
    }
  }
}


filter {
  if [type]=="n-prod" {
    grok {
    match => [ "message", "%{TIMESTAMP_ISO8601:timestamp}%{SPACE}(?<log_level>WARN|ERROR|FATAL)%{SPACE}%{JAVACLASS:java_class_path}" ]
      add_field => {
        "service" => "n"
        "cluster" => "n"
      }
    }
    date {
      match => [ "timestamp" , "YYYY-MM-dd HH:mm:ss,SSS" ]
    }
  }
}


filter {
  if [type]=="b" {

    grok {
    match => [ "message", "%{TIMESTAMP_ISO8601:timestamp}%{SPACE}(?<log_level>WARN|ERROR|FATAL)%{SPACE}%{JAVACLASS:java_class_path}" ]
      add_field => {
        "service" => "b"
        "cluster" => "b"
      }
    }
    date {
      match => [ "timestamp" , "YYYY-MM-dd HH:mm:ss,SSS" ]
    }
  }
}

filter {
  if [type]=="q" {
    grok {
    match => [ "message", "%{TIMESTAMP_ISO8601:timestamp}%{SPACE}(?<log_level>WARN|ERROR|FATAL)%{SPACE}%{JAVACLASS:java_class_path}" ]
      add_field => {
        "service" => "q"
        "cluster" => "q"
      }
    }
    date {
      match => [ "timestamp" , "YYYY-MM-dd HH:mm:ss,SSS" ]
    }
  }
}


filter {
  if [type]=="r" or [type]=="r" {
    multiline {
      pattern => "^=(CRASH REPORT|WARNING REPORT|SUPERVISOR REPORT|ERROR REPORT)=*%{SPACE}%{MONTHDAY}-%{MONTH}-%{YEAR}::%{TIME}"
      negate => true
      what => previous
    }
  
    grok {  
      match => [ "message", "^=(?<log_level>CRASH REPORT|WARNING REPORT|SUPERVISOR REPORT|ERROR REPORT)=*%{SPACE}(?<timestamp>%{MONTHDAY}-%{MONTH}-%{YEAR}::%{TIME})" ]

      add_field => {
        "service" => "r"
        "cluster" => "r"
      }
    }
    
    date {
      match => [ "timestamp" , "d-MMM-YYYY::HH:mm:ss" ]
    }
  }
}




filter {
   if [type]=="e" {  
    multiline {
      pattern => "^\[%{TIMESTAMP_ISO8601}"
      negate => true
      what => previous
    }
    
    grok {  
        match => [ "message", "^\[%{TIMESTAMP_ISO8601:timestamp}\]\[(?<log_level>INFO|DEBUG|ERROR|FATAL|WARN)%{SPACE}*\]\[%{JAVAFILE:elasticsearch_service}"]
      
      add_field => {
        "service" => "e"
        "cluster" => "e"
        }
    }
    date {
      match => [ "timestamp" , "YYYY-MM-dd HH:mm:ss,SSS" ]
    }
  }
}

output {
  if "_grokparsefailure" in [tags] {
    stdout { codec => rubydebug }
  }
  else {
    elasticsearch {
      host => '127.0.0.1'
      cluster => logstash
      bind_port => 9302
    }
  }
}

**Comment by drocsid** [comment]
sorry I forgot to use md

**Comment by udaivizury** [comment]
Not 100% relevant but it might help.
I had same errors flooding my logstash.log file. But when I added the cluster => 'myelasticsearch' in the output file, it started to work. No errors thrown. Following is the config, might help.

My setup is a single node ES cluster, standard port, 1 Shard, no replica
Logstash config - output
output {
  elasticsearch { 
    host => 'x.x.x.x'
    cluster => 'myelasticsearch'
  }
  stdout { codec => rubydebug }
}
====
Logstash config - input
input {
  lumberjack {
    port => 5000
    type => "logs"
    ssl_certificate => "/etc/pki/newtls/certs/logstash-forwarder.crt"
    ssl_key => "/etc/pki/newtls/private/logstash-forwarder.key"
  }
}
=============
On the client node, /etc/logstash-forwarder.conf has
{
  "network": {
    "servers": [ "x.x.x.x:5000" ],
    "ssl certificate": "/etc/pki/newtls/certs/logstash-forwarder.crt",
    "ssl key": "/etc/pki/newtls/private/logstash-forwarder.key",
    "ssl ca": "/etc/pki/newtls/certs/logstash-forwarder.crt",
    "protocol": "transport",
    "timeout": 150
  },

  "files": [
    {
      "paths": [
        "/var/log/syslog"
      ],

      "fields": { "type": "syslog" }
    }
  ]
}
==================

**Comment by censored** [comment]
Using a very generic config - no filters, just connections, I ran into the error in the beginning of the thread. My solution was to set up logstash to reference cluster => "mycluster", matching the cluster.name I defined in elastic search.

**Comment by vlaadbrain** [comment]
logstash-2.0.0 with the following output config

#103 This seems related to the original topic of the issue, i don't understand why its closed.

    elasticsearch {
        action => "index"
        hosts => ["host1", "host2", "host3"]
        document_id => "%{anonid}"
        index => "logstash-%{type}-%{+YYYY.MM.dd}"
        max_retries => 10
        retry_max_interval => 5
        retry_max_items => 100
        flush_size => 100
        timeout => 25
        workers => 10
    }

gets me an error like so and the process crashes...

    Manticore::SocketTimeout: Read timed out
       initialize at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.4-java/lib/manticore/response.rb:35
             call at org/jruby/RubyProc.java:271
             call at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.4-java/lib/manticore/response.rb:70
        call_once at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.4-java/lib/manticore/response.rb:245
             code at /opt/logstash/vendor/bundle/jruby/1.9/gems/manticore-0.4.4-java/lib/manticore/response.rb:148
    perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.14/lib/elasticsearch/transport/transport/http/manticore.rb:71
             call at org/jruby/RubyProc.java:271
    perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.14/lib/elasticsearch/transport/transport/base.rb:191
    perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.14/lib/elasticsearch/transport/transport/http/manticore.rb:54
    perform_request at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-transport-1.0.14/lib/elasticsearch/transport/client.rb:119
             bulk at /opt/logstash/vendor/bundle/jruby/1.9/gems/elasticsearch-api-1.0.14/lib/elasticsearch/api/actions/bulk.rb:87
             bulk at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.1.2-java/lib/logstash/outputs/elasticsearch/http_client.rb:56
           submit at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.1.2-java/lib/logstash/outputs/elasticsearch.rb:353
      synchronize at org/jruby/ext/thread/Mutex.java:149
           submit at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.1.2-java/lib/logstash/outputs/elasticsearch.rb:350
      retry_flush at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.1.2-java/lib/logstash/outputs/elasticsearch.rb:514
         register at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-2.1.2-java/lib/logstash/outputs/elasticsearch.rb:294

**Comment by andrewvc** [comment]
@vlaadbrain we're taking a look right now. The reason the original issue is closed is that this thread became a dumping ground for errors with a similar message but different underlying causes. I hope you understand. Thank you for reporting this bug. Would you mind opening it as a separate issue and linking to it from here?

**Comment by vlaadbrain** [comment]
Hi Andrew, i've created #304 for tracking

Thanks

**Comment by sammcj** [comment]
Looks like this might be the same issue as in #2241
